File: src/telemetry/python/telemetry.py
"""
Local-only telemetry (public safe)
"""

import time

class Telemetry:
    def __init__(self):
        self.events = []

    def record(self, event: str, data=None):
        self.events.append({
            "event": event,
            "data": data or {},
            "ts": int(time.time()*1000)
        })

    def flush(self):
        batch = list(self.events)
        self.events.clear()
        return batch

File: src/telemetry/js/telemetry.js
/**
 * Local-only telemetry module (public safe)
 */
export class Telemetry {
  constructor() {
    this.events = [];
  }

  record(event, data = {}) {
    this.events.push({
      event,
      data,
      ts: Date.now()
    });
  }

  flush() {
    const batch = [...this.events];
    this.events = [];
    return batch;
  }
}
